crates/cxx-qt-gen/src/generator/rust/signals.rs

```diff
@@ -11,11 +11,137 @@ use crate::{
+        parse_quote! { &#qobject_name }
+    };
+    let self_type_qualified =
+        syn_type_cxx_bridge_to_qualified(&self_type_cxx, &cxx_mappings.qualified);
+    let qualified_impl =
+        syn_ident_cxx_bridge_to_qualified_impl(qobject_name, &cxx_mappings.qualified);
+
+    let mut unsafe_block = None;
+    let mut unsafe_call = Some(quote! { unsafe });
+    if signal.safe {
+        std::mem::swap(&mut unsafe_call, &mut unsafe_block);
+    }
+
+    let attrs = &signal.method.attrs;
+
+    let fragment = RustFragmentPair {
+        cxx_bridge: vec![
+            quote! {
+                #unsafe_block extern "C++" {
+                    #(#attrs)*
+                    #[rust_name = #signal_name_rust_str]
+                    #unsafe_call fn #signal_name_cpp(self: #self_type_cxx, #(#parameters_cxx),*);
+                }
+            },
+            quote! {
+                unsafe extern "C++" {
+                    #[doc(hidden)]
+                    #[namespace = #connect_namespace]
+                    #[must_use]
+                    #[rust_name = #free_connect_ident_rust_str]
+                    fn #free_connect_ident_cpp(self_value: #self_type_cxx, func: #unsafe_call fn(#self_type_cxx, #(#parameters_cxx),*), conn_type: CxxQtConnectionType) -> CxxQtQMetaObjectConnection;
+                }
+            },
+        ],
+        implementation: vec![
+            quote! {
+                impl #qualified_impl {
+                    #[doc = "Connect the given function pointer to the signal "]
+                    #[doc = #signal_name_cpp_str]
+                    #[doc = ", so that when the signal is emitted the function pointer is executed."]
+                    #[doc = "\n"]
+                    #[doc = "Note that this method uses a AutoConnection connection type."]
+                    #[must_use]
+                    pub fn #on_ident_rust(self: #self_type_qualified, func: fn(#self_type_qualified, #(#parameters_qualified),*)) -> cxx_qt_lib::QMetaObjectConnection
+                    {
+                        #module_ident::#free_connect_ident_rust(self, func, cxx_qt_lib::ConnectionType::AutoConnection)
+                    }
+                }
+            },
+            quote! {
+                impl #qualified_impl {
+                    #[doc = "Connect the given function pointer to the signal "]
+                    #[doc = #signal_name_cpp_str]
+                    #[doc = ", so that when the signal is emitted the function pointer is executed."]
+                    #[must_use]
+                    pub fn #connect_ident_rust(self: #self_type_qualified, func: fn(#self_type_qualified, #(#parameters_qualified),*)) -> cxx_qt_lib::QMetaObjectConnection
+                    {
+                        #module_ident::#free_connect_ident_rust(self, func, cxx_qt_lib::ConnectionType::AutoConnection)
+                    }
+                }
+            },
+        ],
+    };
+
+    let mut generated = GeneratedRustQObject::default();
```

Comment:
Why doesn't this just return a `GeneratedExternCxxQt` directly?

Reply:
done